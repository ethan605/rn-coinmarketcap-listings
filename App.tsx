/**
 * @format
 */

import React, { PureComponent } from 'react';
import { SafeAreaView, StatusBar } from 'react-native';
import { Provider } from 'react-redux';

// Locals
import PlaceholderScreen from 'src/components/PlaceholderScreen';
import buildStore from 'src/redux';

export default class App extends PureComponent {
  store = buildStore();

  render() {
    return (
      <Provider store={this.store}>
        <StatusBar barStyle="dark-content" />
        <SafeAreaView>
          <PlaceholderScreen />
        </SafeAreaView>
      </Provider>
    );
  }
};
