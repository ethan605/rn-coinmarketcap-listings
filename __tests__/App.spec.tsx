/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
// import { call, put } from 'redux-saga/effects';
// import { SagaIterator } from 'redux-saga';

import App from '../App';

describe('App', (): void => {
  it('renders correctly', (): void => {
    const snapshot = renderer.create(<App />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

// describe('Experimentials', (): void => {
//   function doubler(num: number): Promise<number> {
//     return new Promise<number>(
//       (resolve): void => {
//         console.debug('inside doubler:', num);
//         resolve(num * 2);
//       }
//     );
//   }

//   function* main(num: number): SagaIterator {
//     const val = yield call(doubler, num);
//     yield put({ type: 'TEST:SUCCESS', payload: val });
//   }

//   function incrementor(otherGenerator: (num: number) => SagaIterator): (num: number) => SagaIterator {
//     function* generator(num: number): SagaIterator {
//       yield put({ type: 'TEST', payload: num + 1 });
//       yield* otherGenerator(num);
//     }

//     return generator;
//   }

//   const generator = incrementor(main)(3);
//   console.debug(generator.next().value);
//   console.debug(generator.next().value);
//   console.debug(generator.next().value);
// });
