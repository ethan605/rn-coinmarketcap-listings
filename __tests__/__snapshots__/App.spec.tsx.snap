// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      Object {
        "backgroundColor": "lightgray",
      }
    }
  >
    <View>
      <Text>
        CoinsMarketCap - Listings
      </Text>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
